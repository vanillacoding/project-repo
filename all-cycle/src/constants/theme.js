const THEME = {
  white: {
    title: "white",
    color: "#FEFFFF",
  },
  lightGray: {
    title: "light gray",
    color: "#F6F7F7",
  },
  gray: {
    title: "icon color",
    color: "#A69E9E",
  },
  primary: {
    title: "primary color",
    color: "#3DD97E",
  },
  green: {
    title: "main color1",
    color: "#25AD5F",
  },
  graishGreen: {
    title: "main color2",
    color: "#59A380",
  },
  darkGreen: {
    title: "main color3",
    color: "#087D3D",
  },
  lightGreen: {
    title: "light green",
    color: "#ADD999",
  },
  yellow: {
    title: "yellow",
    color: "#FFC914",
  },
  darkGray: {
    title: "dark gray font",
    color: "#707070",
  },
  lightFont: {
    title: "light font color",
    color: "#5C5C5C",
  },
  font: {
    title: "font color",
    color: "#414141",
  },
  orange: {
    title: "orange",
    color: "#F19949",
  },
  red: {
    title: "red",
    color: "#F06449",
  },
  skyblue: {
    title: "skyblue",
    color: "#60C4F9",
  },
  badgeBg: {
    title: "badge background color",
    color: "#E1E1E1",
  },
  fontEng: "Montserrat, sans-serif",
  fontKor: "Nanum Gothic, sans-serif",
};

export default THEME;
