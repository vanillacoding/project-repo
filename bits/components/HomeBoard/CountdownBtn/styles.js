import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  pressButtonWrapper: {
    justifyContent: 'space-evenly',
    alignItems: 'center',
    flex: 1
  },
  pressButtonText: {
    fontWeight: '700',
    fontSize: 17,
    color: 'white'
  },
  circleWrapper: {
    marginTop: 30
  },
  doneText: {
    color: '#4cd137',
    fontWeight: '600'
  },
  btnWrapper: {
    flexDirection: 'row',
  },
  pauseBtn: {
    marginRight: 15
  },
  cancelBtn: {
    marginLeft: 15
  }
});

export default styles;
