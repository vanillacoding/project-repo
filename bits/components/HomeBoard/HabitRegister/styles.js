import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  mainTopWrapper: {
    flexGrow: 0.4,
    justifyContent: 'center',
    alignItems: 'center'
  },
  habitRegisterWrapper: {
    backgroundColor: '#EDCD88',
    justifyContent: 'center',
    alignItems: 'center',
    width: '80%',
    height: '25%',
    borderRadius: 10,
    flexGrow: 0.8,
  },
  textButtonWrapper: {
    height: '100%',
    justifyContent: 'space-evenly',
    alignItems: 'center'
  },
  registerText: {
    fontWeight: '900',
    color: 'white',
    fontSize: 15
  }
});

export default styles;
