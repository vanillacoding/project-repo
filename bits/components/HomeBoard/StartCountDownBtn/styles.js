import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  wrapper: {
    justifyContent: 'center',
    alignItems: 'center',
    height: '44%'
  },
  outterCircle: {
    borderWidth: 13,
    width: 170,
    height: 170,
    borderRadius: 100,
    borderColor: 'white',
    justifyContent: 'center',
    alignItems: 'center'
  },
  innerCircle: {
    width: 140,
    height: 140,
    borderRadius: 100,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F4DFB0'
  },
  pressText: {
    fontSize: 15,
    fontWeight: '600'
  }
});

export default styles;
