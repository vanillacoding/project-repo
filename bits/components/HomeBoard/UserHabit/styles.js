import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  wrapper: {
    justifyContent: 'center',
    alignItems: 'center',
    height: 200
  },
  userHabitWrapper: {
    width: '80%',
    height: '80%',
    borderRadius: 10,
    backgroundColor: '#FACC7D'
  },
  habitInfoWrapper: {
    flexDirection: 'row',
    justifyContent: 'space-evenly',
    alignItems: 'center',
    width: 255
  },
  iconWrapper: {
    backgroundColor: '#F9BC56',
    borderRadius: 10,
    padding: 20,
    marginLeft: 10,
  },
  habitInfo: {
    height: '80%',
    justifyContent: 'space-evenly',
    marginLeft: 20
  },
  name: {
    color: 'white',
    fontWeight: '900',
    fontSize: 30
  },
  dayMateLike: {
    justifyContent: 'space-evenly',
    height: '60%'
  },
  day: {
    color: 'white',
    fontWeight: '600',
    fontSize: 15
  },
  mate: {
    color: 'white',
    fontWeight: '600',
    fontSize: 15
  },
  like: {
    color: 'white',
    fontWeight: '600',
    fontSize: 15
  },
  status: {
    color: 'white',
    fontWeight: '600',
    fontSize: 15
  },
  cancelWrapper: {
    justifyContent: 'flex-start',
    height: '90%',
    marginLeft: 10
  },
  addBtnWrapper: {
    top: 15
  }
});

export default styles;
