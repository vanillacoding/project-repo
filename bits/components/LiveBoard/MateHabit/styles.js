import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  mateHabitWrapper: {
    flex: 1,
    width: '100%',
    backgroundColor: 'white'
  },
  mateLivewrapper: {
    paddingTop: 10,
    paddingBottom: 10
  },
  profileImg: {
    flexDirection: 'row',
    alignItems: 'center'
  },
  uriWrapper: {
    marginLeft: 10,
    paddingRight: 10
  },
  profileUriImg: {
    width: 40,
    height: 40,
    borderRadius: 100,
    borderWidth: 1,
    borderColor: 'gray'
  },
  img: {
    marginLeft: 3,
    paddingRight: 7
  },
  name: {
    marginLeft: 4,
    fontWeight: '500'
  },
  habitWrapper: {
    backgroundColor: '#E8C06C',
    margin: 8,
    borderRadius: 10,
    width: 100,
    height: 100,
    justifyContent: 'space-evenly',
    alignItems: 'center'
  },
  statusTextWrapper: {
    width: '100%'
  },
  statusText: {
    textAlign: 'right',
    color: 'white',
    marginRight: 10,
    fontSize: 10
  },
  habitIcon: {
    justifyContent: 'center',
    alignItems: 'center'
  },
  startTimeText: {
    color: 'white',
    fontWeight: '800'
  }
});

export default styles;
