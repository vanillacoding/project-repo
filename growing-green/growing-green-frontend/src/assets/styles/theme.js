const baseTheme = {
  fontSizes: ['1em', '1.1em', '1.2em', '1.3em', '2em'],
  colors: {
    blue: '#07c',
    tomato: 'tomato',
    purple: 'purple',
    white: '#fcfcfc',
    darkGreen: '#457f51',
    green: '#9cbd93',
    lightGreen: 'rgba(178, 219, 177, 0.49)',
    black: '#111',
    gray: '#444',
    lightGray: 'rgba(208, 208, 208, 0.3)',
    red: '#b12f35',
    darkRed: '#892429',
    ivory: '#fbfbf9',
    lightBlack: 'rgba(0, 0, 0, 0.35)',
  },
};

const buttons = {
  outline: {
    borderStyle: 'solid',
    borderWidth: '1px',
    width: '140px',
    height: '20px',
    borderRadius: 'none',
  },
  rounded: {
    borderStyle: 'solid',
    borderWidth: '2px',
    width: '200px',
    height: '16px',
    borderRadius: '16px',
  },
};

const buttonSizes = {
  small: {
    width: '110px',
    height: '11px',
    fontSize: baseTheme.fontSizes[0],
    fontWeight: '300',
    lineHeight: '20px',
    padding: '13px 15px',
  },
  medium: {
    width: '140px',
    height: '14px',
    fontSize: baseTheme.fontSizes[1],
    fontWeight: '300',
    lineHeight: '13px',
    padding: '14px 17px',
  },
  large: {
    width: '170px',
    height: '20px',
    fontSize: baseTheme.fontSizes[2],
    fontWeight: '600',
    lineHeight: '20px',
    padding: '15px 15px',
  },
  long: {
    width: '300px',
    height: '15px',
    fontSize: baseTheme.fontSizes[2],
    fontWeight: '600',
    lineHeight: '20px',
    padding: '10px 15px',
  },
  short: {
    width: '180px',
    height: '26px',
    fontSize: baseTheme.fontSizes[2],
    fontWeight: '600',
    lineHeight: '20px',
    padding: '8px 15px',
  },
};

const buttonColors = {
  white: {
    background: baseTheme.colors.white,
    text: baseTheme.colors.black,
    shadow: baseTheme.colors.gray,
    border: baseTheme.colors.black,
  },
  green: {
    background: baseTheme.colors.ivory,
    text: baseTheme.colors.black,
    shadow: baseTheme.colors.green,
    border: baseTheme.colors.black,
  },
  red: {
    background: baseTheme.colors.ivory,
    text: baseTheme.colors.black,
    shadow: baseTheme.colors.red,
    border: baseTheme.colors.black,
  },
  translucentGreen: {
    background: 'rgb(88, 126, 78, 0.5)',
    text: baseTheme.colors.black,
    border: baseTheme.colors.green,
  },
  translucentRed: {
    background: 'rgb(204, 0, 0, 0.3)',
    text: baseTheme.colors.red,
    border: baseTheme.colors.red,
  },
};

module.exports = {
  baseTheme,
  buttonSizes,
  buttonColors,
  buttons,
};
