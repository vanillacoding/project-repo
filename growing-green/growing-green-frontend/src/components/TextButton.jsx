import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { buttons, buttonSizes, buttonColors } from '../assets/styles/theme';

export default function TextButton({ ...props }) {
  return (
    <StyledButton {...props}>
      {props.icon && <props.icon size="24" />}
      {props.label}
    </StyledButton>
  );
}

TextButton.propTypes = {
  onClick: PropTypes.func,
  color: PropTypes.string,
  icon: PropTypes.func,
  size: PropTypes.string,
  label: PropTypes.string.isRequired,
};

TextButton.defaultProps = {
  type: 'button',
  size: 'large',
  color: 'white',
};

const StyledButton = styled.button`
  all: unset;
  display: flex;
  justify-content: center;
  align-items: center;
  cursor: pointer;
  box-shadow: ${({ variant, color }) =>
    variant === 'outline'
      ? `-5px 5px 0px 0px ${buttonColors[color].shadow}`
      : 'none'};
  border-style: solid;
  border-width: ${({ variant }) => buttons[variant].borderWidth};
  border-color: ${({ color }) => buttonColors[color].border};
  width: ${({ size }) => buttonSizes[size].width};
  height: ${({ size }) => buttonSizes[size].height};
  padding: ${({ size }) => buttonSizes[size].padding};
  font-size: ${({ size }) => buttonSizes[size].fontSize};
  line-height: ${({ size }) => buttonSizes[size].lineHeight};
  font-weight: ${({ size }) => buttonSizes[size].fontWeight};
  color: ${({ color }) => buttonColors[color].text};
  border-radius: ${({ variant, size }) =>
    variant === 'rounded' && buttonSizes[size].height};
  background-color: ${({ color }) => buttonColors[color].background};
  &:active {
    box-shadow: none;
    ${({ variant }) =>
      variant === 'outline' ? `transform: translate(-5px)` : 'transform: none'};
  }

  svg {
    margin-right: 1.2rem;
  }
`;
