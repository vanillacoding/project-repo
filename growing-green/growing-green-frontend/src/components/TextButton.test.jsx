import React from 'react';
import TextButton from './TextButton';
import 'jest-styled-components';
import { render, fireEvent } from '../test/test-utils';
import { buttonSizes, buttonColors, buttons } from '../assets/styles/theme';

describe.only('<TextButton>', () => {
  const sampleText = 'sampleButton';

  it('matches snapshot', () => {
    const { container } = render(
      <TextButton
        onClick={() => {}}
        variant="rounded"
        size="short"
        color="translucentRed"
        label={sampleText}
      />,
    );

    expect(container).toMatchSnapshot();
  });

  it('should have style element according to the given attribute value', () => {
    const size = 'short';
    const color = 'translucentRed';
    const variant = 'rounded';

    const { getByRole } = render(
      <TextButton
        onClick={() => {}}
        variant={variant}
        size={size}
        color={color}
        label={sampleText}
      />,
    );

    expect(getByRole('button')).toHaveStyleRule(
      'width',
      buttonSizes[size].width,
    );
    expect(getByRole('button')).toHaveStyleRule(
      'height',
      buttonSizes[size].height,
    );
    expect(getByRole('button')).toHaveStyleRule(
      'color',
      buttonColors[color].text,
    );
    expect(getByRole('button')).toHaveStyleRule(
      'color',
      buttonColors[color].text,
    );
    expect(getByRole('button')).toHaveStyleRule(
      'border-width',
      buttons[variant].borderWidth,
    );
  });

  it('should event occurs when the user clicks the button', () => {
    jest.spyOn(global.console, 'log');

    const { getByText } = render(
      <TextButton
        onClick={() => console.log('click')}
        variant="rounded"
        size="short"
        color="translucentRed"
        label={sampleText}
      />,
    );

    const button = getByText(sampleText);

    fireEvent.click(button);

    expect(console.log).toHaveBeenCalledWith('click');
  });
});
