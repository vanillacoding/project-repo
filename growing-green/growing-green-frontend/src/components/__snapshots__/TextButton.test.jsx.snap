// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextButton> matches snapshot 1`] = `
.c0 {
  all: unset;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  box-shadow: none;
  border-style: solid;
  border-width: 2px;
  border-color: #b12f35;
  width: 180px;
  height: 26px;
  padding: 8px 15px;
  font-size: 1.2em;
  line-height: 20px;
  font-weight: 600;
  color: #b12f35;
  border-radius: 26px;
  background-color: rgb(204,0,0,0.3);
}

.c0:active {
  box-shadow: none;
  -webkit-transform: none;
  -ms-transform: none;
  transform: none;
}

.c0 svg {
  margin-right: 1.2rem;
}

<div>
  <button
    class="c0"
    color="translucentRed"
    label="sampleButton"
    type="button"
  >
    sampleButton
  </button>
</div>
`;
